### PR TITLE
glib: fix gdbus-codegen

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **glib/***

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
this intends to solve [this error](https://github.com/bincrafters/conan-gtk/runs/1188800537#step:5:2667):
```
  File "/home/conan/.conan/data/glib/2.66.0/_/_/package/9fd9dea5f5135300c47d42593722bb502d28d66e/bin/gdbus-codegen", line 53, in <module>
    from codegen import codegen_main
ModuleNotFoundError: No module named 'codegen'
```

File: recipes/glib/all/test_package/conanfile.py [hunk 1]
```diff
@@ -7,6 +7,9 @@ class TestPackageConan(ConanFile):
     generators = "cmake", "cmake_find_package", "pkg_config"
 
     def build(self):
+        if self.settings.os != 'Windows':
+            self.run('gdbus-codegen -h', run_environment=True)
```
Comment (madebr) [inline recipes/glib/all/test_package/conanfile.py:11]: Can this script run on winbash?

Comment (ericLemanissier) [reply recipes/glib/all/test_package/conanfile.py:11]: probably

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@uilianries  @danimtb the bot running on master indicates that the merge commit integrated properly : https://github.com/conan-io/conan-center-index/commit/3f54c4aecbbb730d6a963f00c58706527c58a880
However the recipe and binaries do not seem to be updated yet (https://bintray.com/conan/conan-center/glib%3A_/2.66.1%3A_ and https://conan.io/center/glib?tab=recipe)
What's happening ?

**Comment by danimtb** [comment]
I see the package there: https://bintray.com/conan/conan-center/glib%3A_/2.66.1%3A_#files/_/glib/2.66.1/_/4a52b8df69077afdf5108cba99c29438/package

Recipe at https://conan.io/center/glib?tab=recipe seems updated to me.

Also, the revision for other versions seems updated (21 of October) in Bintray:

```
λ conan search glib/2.65.0@ -r conan-center --revisions
Revisions for 'glib/2.65.0' at remote 'conan-center':
49aa1b7230fa97d1e34184ab37a7353b (2020-10-21 03:53:01 UTC)
0bcf215abee44f3f05490bf2faec688c (2020-10-15 18:33:40 UTC)
19c3faa62f5268334fcd1c565d64b142 (2020-10-14 18:34:09 UTC)
26c5b6d99d91b3dfe1eddcf6285bcae6 (2020-09-27 15:34:59 UTC)
093a01591f6a33fa0ee321a6ed8c4c4b (2020-09-26 16:03:58 UTC)
244fe900ab2991d7e44c85a9514764ef (2020-09-21 11:34:40 UTC)
965412c6a132809691099521a71e1a31 (2020-09-10 14:08:40 UTC)
7a4ed52671fcfe521c860e30b185f20f (2020-08-10 08:20:00 UTC)
0af4494c1a0a8880dc5e09863e904814 (2020-07-29 12:33:56 UTC)
4dc4415be2a87811e340945103ca6993 (2020-07-28 08:02:45 UTC)
ecc6e90378f2ed063a2e3b4004b7d0b1 (2020-07-27 15:53:04 UTC)
eabb4ba68fabe8f66915264ae4f7b655 (2020-07-03 09:32:00 UTC)
```

Sometimes the new revisions of packages got queued in the backend and the web UI may take a little time to update. However, the packages should show up in the Conan client if you search Bintray when packages are uploaded after the merge.

Hope everything looks ok to you as well :)

**Comment by ericLemanissier** [comment]
https://conan.io/center/glib?tab=recipe does not have this PR: there is no occurrence of "codegen" in it
both revisions in https://bintray.com/conan/conan-center/glib%3A_/2.66.1%3A_#files/_/glib/2.66.1/_ are too old to be this PR (6 and 7 days)
`conan search glib/2.66.1@ -r conan-center --revisions` returns 
```
Revisions for 'glib/2.66.1' at remote 'conan-center':
4a52b8df69077afdf5108cba99c29438 (2020-10-15 18:57:18 UTC)
ccf13ce211cf7f41503cc8818dd1344f (2020-10-14 19:02:40 UTC)
```
So the problem seems to be limited to 2.66.1

**Comment by danimtb** [comment]
Apparently, this PR conflicted with the changes in https://github.com/conan-io/conan-center-index/pull/3190 but this PR (#3063) was not updated with those latest changes in master, so version 2.66.1 was never built here. Once the conflict was detected in master, the job properly built the conflicting ones, but 2.66.1 was not in repo so it was left as it was.

Certainly, this is a bug in the merge pipeline.

**Comment by ericLemanissier** [comment]
My bad too, I forgot that I had this PR when submitting #3063. I would happily welcome a bot which puts a comment on each new PR on a recipe which already has a PR in flight (even if there is no conflict)

**Comment by ericLemanissier** [comment]
@danimtb should I create another "regenerate glib" PR, or can you somehow recreate the missing 2.66.1 packages ?

**Comment by jgsogo** [comment]
> My bad too, I forgot that I had this PR when submitting #3063. I would happily welcome a bot which puts a comment on each new PR on a recipe which already has a PR in flight (even if there is no conflict)

I think it is ok to have several PRs to the same recipe at the same time (otherwise PRs might get bigger and complex), but the bot should realize that the PR is not _rebased_ (only that recipe, not the full repository) before merging it. I think this is important not just because of this error, but because two PRs can be incompatible even if there is no git-conflict between them.... and we don't want to know about this error after changes are merged.

---

Yes, please, open a dummy PR to `glib` to _regenerate_ that version. We are preparing a job to fill these gaps automatically, but it is not ready yet.

Thanks!

**Comment by madebr** [comment]
What is the bots behavior when the package fails to build after a successful merge/rebase?

**Comment by jgsogo** [comment]
> What is the bots behavior when the package fails to build after a successful merge/rebase?

Right now: the bot asks for help. 
 * If the package fails to build and changes in the recipe are needed, then a new PR is the way to go.
 * If it is a _random_ failure, we trigger again that job (the job is idempotent)

**Comment by ericLemanissier** [comment]
I'm not saying that having several PR on the same package is always a problem, but usually you want to know it. It would have saved me some time on several occasions
I opened a [PR for glib 2.66.2](https://github.com/conan-io/conan-center-index/pull/3290), but does the bot regenerate previous versions, even when the PR is identified as "Bump version" ?

**Comment by madebr** [comment]
> > What is the bots behavior when the package fails to build after a successful merge/rebase?
> 
> Right now: the bot asks for help.
> 
>     * If the package fails to build and changes in the recipe are needed, then a new PR is the way to go.
> 
>     * If it is a _random_ failure, we trigger again that job (the job is idempotent)

What I meant to ask is:
e.g. https://github.com/conan-io/conan-center-index/pull/3063#issuecomment-702197311
Did the bot test commit 32c92245b7dc176032e4c3a7a002825fc41d6f93 or the merge commit with current master?

**Comment by jgsogo** [comment]
> I opened a [PR for glib 2.66.2](https://github.com/conan-io/conan-center-index/pull/3290), but does the bot regenerate previous versions, even when the PR is identified as "Bump version" ?

Yes, in this scenario, the pipeline will detect that the recipe revision is not the same after merging to `master` and it will compile all the packages again.

**Comment by jgsogo** [comment]
> What I meant to ask is:
> e.g. [#3063 (comment)](https://github.com/conan-io/conan-center-index/pull/3063#issuecomment-702197311)
> Did the bot test commit [32c9224](https://github.com/conan-io/conan-center-index/commit/32c92245b7dc176032e4c3a7a002825fc41d6f93) or the merge commit with current master?

The pipeline compiles the sources after merging (locally) to `master`, not the sources of the commit itself. The problem can arise if some other PR is merged to `master` after running this one and before merging this one.

**Pull request merged**